lib/gmputil.h

```diff
@@ -84,7 +84,7 @@ static inline unsigned bitcount(big_int v) {
 }
 
 static inline int ffs(big_int v) {
-    if (v <= 0) return -1;
+    if (v <= 0) return (v < 0) ? -1 : 0;
```

Comment:
we should probably not change the definition of ffs; if all bits are 0 returning 1 seems incorrect. Instead probably you should move this check to the function that calls ffs.